### PR TITLE
[PLAYER-556] Reworked to accept all player parameters

### PR DESCRIPTION
**Pull request by mityaha**

### CODE REVIEW

File: amp_iframe.html [hunk 1]
```diff
@@ -4,67 +4,76 @@
...
-    // Parse docUrl
-    var docUrl = undefined;
-    if (!!queryParams.match(/docUrl=([^&]*)/)) {
-      playerParam["docUrl"] = decodeURIComponent(queryParams.match(/docUrl=([^&]*)/)[1]);
-    }
+    function queryStringToJSON(qs) {
+      var pairs = qs.split('&');
+      var result = {};
+      pairs.forEach(function(pair) {
+        var pair = pair.split('=');
```
Comment (mistermendez) [inline amp_iframe.html]: should `var pair` have another name? the argument to the function is also `pair`

File: amp_iframe.html [hunk 2]
```diff
@@ -4,67 +4,76 @@
...
+            result[key].push(value);
+          } else {
+            result[key] = [result[key], value];
           }
-          if (typeof currentLevel !== 'object')
-            throw("Exception: Invalid query parameter");
+        } else {
+          result[key] = value;
+        }
+      });
```
Comment (mistermendez) [inline amp_iframe.html]: semicolon not needed here

File: iframe.html [hunk 3]
```diff
@@ -13,62 +13,66 @@
...
-    } else if (document.referrer) {
-      playerParam["docUrl"] = document.referrer;
-    } else {
-      console.log("iframe.html - document.referrer was not set");
-    }
+    function queryStringToJSON(qs) {
+      var pairs = qs.split('&');
+      var result = {};
+      pairs.forEach(function(pair) {
+        var pair = pair.split('=');
```
Comment (mistermendez) [inline iframe.html]: should `var pair` have another name? the argument to the function is also `pair`

File: iframe.html [hunk 4]
```diff
@@ -13,62 +13,66 @@
...
-            currentLevel[keySplits[0]] = true;
-          else if (paramValue.toLowerCase() === "false")
-            currentLevel[keySplits[0]] = false;
-          else if (paramValue[0] === "[" &&
-            paramValue[paramValue.length -1] === "]")
-            currentLevel[keySplits[0]] = JSON.parse(decodeURIComponent(match[2]));
-          else
-            currentLevel[keySplits[0]] = decodeURIComponent(match[2]);
+      return JSON.parse(JSON.stringify(result));
+    };
```
Comment (mistermendez) [inline iframe.html]: semicolon not needed here

File: amp_iframe.html [hunk 5]
```diff
@@ -4,67 +4,76 @@
...
+        console.log("iframe.html - document.referrer was not set");
       }
     }
 
     OO.ready(function() {
-      window.pp = OO.Player.create('container', queryParams.match(/ec=([^&]*)/)[1], playerParam);
+      window.pp = OO.Player.create('container', playerParams.ec, playerParams);
 
-      //Enable cross-iframe control
+        //Enable cross-iframe control
```
Comment (mistermendez) [inline amp_iframe.html]: extra space not needed

File: amp_iframe.html [hunk 6]
```diff
@@ -4,67 +4,76 @@
...
-      playerParam["docUrl"] = decodeURIComponent(queryParams.match(/docUrl=([^&]*)/)[1]);
-    }
+    function queryStringToJSON(qs) {
+      var pairs = qs.split('&');
+      var result = {};
+      pairs.forEach(function(pair) {
+        var pair = pair.split('=');
+        var key = pair[0];
+        var value = decodeURIComponent(pair[1] || '');
+        if (value === "true") {
```
Comment (mistermendez) [inline amp_iframe.html]: would switch statement work better than if/else here? 

the logic is the same but switch is more efficient than else/if. see - http://stackoverflow.com/questions/449273/why-the-switch-statement-and-not-if-else

File: amp_iframe.html [hunk 7]
```diff
@@ -4,67 +4,76 @@
...
+            source: "<iframe width='640' height='480' frameborder='0' allowfullscreen src='//player.ooyala.com/static/v4/stable/latest/skin-plugin/iframe.html?ec=<ASSET_ID>&pbid=<PLAYER_ID>&pcode=<PUBLISHER_ID>'></iframe>"
+          }
         }
       }
-      catch (e) {
-        throw("Exception: Invalid query parameter");
+    };
+    playerParams["playerBrandingId"] = playerParams.pbid;
+    playerParams["skin"] = skin;
+    delete playerParams.pbid;
```
Comment (mistermendez) [inline amp_iframe.html]: why do we delete here?

File: iframe.html [hunk 8]
```diff
@@ -13,62 +13,66 @@
...
+            source: "<iframe width='640' height='480' frameborder='0' allowfullscreen src='//player.ooyala.com/static/v4/stable/latest/skin-plugin/iframe.html?ec=<ASSET_ID>&pbid=<PLAYER_ID>&pcode=<PUBLISHER_ID>'></iframe>"
+          }
         }
       }
-      catch (e) {
-        throw("Exception: Invalid query parameter");
+    };
+    playerParams["playerBrandingId"] = playerParams.pbid;
+    playerParams["skin"] = skin;
+    delete playerParams.pbid;
```
Comment (mistermendez) [inline iframe.html]: why do we delete here?

Comment (mityaha) [inline iframe.html]: We may leave it, but this will be unused parameter, it's cleaner this way I believe.

File: amp_iframe.html (hunk 2 above)
Comment (mityaha) [inline amp_iframe.html]: Why not?

File: amp_iframe.html (hunk 6 above)
Comment (mityaha) [inline amp_iframe.html]: It doesn't look to me that switch here is feasible as these are some of conditions are logical, not string comparisons.

File: amp_iframe.html (hunk 1 above)
Comment (mityaha) [inline amp_iframe.html]: Good point, fixed

File: iframe.html (hunk 3 above)
Comment (mityaha) [inline iframe.html]: Fixed

File: iframe.html (hunk 4 above)
Comment (mityaha) [inline iframe.html]: Fixed

File: amp_iframe.html (hunk 5 above)
Comment (mityaha) [inline amp_iframe.html]: Fixed

File: iframe.html [hunk 9]
```diff
@@ -13,62 +13,66 @@
...
-      console.log("iframe.html - document.referrer was not set");
-    }
+    function queryStringToJSON(qs) {
+      var pairs = qs.split('&');
+      var result = {};
+      pairs.forEach(function(pair) {
+        var pair = pair.split('=');
+        var key = pair[0];
+        var value = decodeURIComponent(pair[1] || '');
+        if (value === "true") {
```
Comment (mityaha) [inline iframe.html]: See my answer in another similar comment

File: amp_iframe.html (hunk 7 above)
Comment (mityaha) [inline amp_iframe.html]: We may leave it, but this will be unused parameter, it's cleaner this way I believe.

File: amp_iframe.html (hunk 2 above)
Comment (mistermendez) [inline amp_iframe.html]: it's a function

File: amp_iframe.html (hunk 7 above)
Comment (mistermendez) [inline amp_iframe.html]: coolio

File: iframe.html (hunk 8 above)
Comment (mistermendez) [inline iframe.html]: okay

**Comment by mistermendez** [comment]
👍

**Pull request merged**